DFC.ServiceTaxonomy.GraphSync/GraphSyncers/Helpers/DescribeContentItemHelper.cs

```diff
@@ -0,0 +1,208 @@
+                    {
+                        context.CurrentDepth = parentContext.CurrentDepth + 1;
+                        var parentRelationship = parentContext.AvailableRelationships.FirstOrDefault(x => x.Destination.All(child.Source.Contains));
+
+                        if (parentRelationship != null && !string.IsNullOrEmpty(parentRelationship.RelationshipPathString))
+                        {
+                            var relationshipString = $"{parentRelationship.RelationshipPathString}-[r{context.CurrentDepth}:{child.Relationship}]-(d{context.CurrentDepth}:{string.Join(":", child.Destination!)})";
+                            child.RelationshipPathString = relationshipString;
+                        }
+                        else
+                        {
+                            context.CurrentDepth = 1;
+                            child.RelationshipPathString = $@"match (s:{string.Join(":", context.SourceNodeLabels)} {{{context.SourceNodeIdPropertyName}: '{context.SourceNodeId}'}})-[r{1}:{child.Relationship}]-(d{1}:{string.Join(":", child.Destination!)})";
+                        }
+                    }
+                }
+
+                currentList.AddRange(context.AvailableRelationships);
+            }
+
+            foreach (var childContext in context.ChildContexts)
+            {
+                await GetRelationships((IDescribeRelationshipsContext)childContext, currentList, context);
+            }
+
+            return currentList;
+        }
+
+        public async Task BuildRelationships(ContentItem contentItem, IDescribeRelationshipsContext context, string sourceNodeIdPropertyName, string sourceNodeId, IEnumerable<string> sourceNodeLabels)
+        {
+            var contentTypeDefinition = _contentDefinitionManager.GetTypeDefinition(contentItem.ContentType);
+
+            _graphSyncHelper.ContentType = contentItem.ContentType;
+
+            var itemContext = contentItem == RootContentItem ? context : new DescribeRelationshipsContext(sourceNodeIdPropertyName, sourceNodeId, sourceNodeLabels, contentItem, _graphSyncHelper, _contentManager, _publishedContentItemVersion, context, _serviceProvider);
+
+            foreach (var part in contentTypeDefinition.Parts)
+            {
+                var partSyncer = _contentPartGraphSyncers.FirstOrDefault(x => x.PartName == part.Name);
+
+                if (partSyncer != null)
+                {
+                    await partSyncer.AddRelationship(itemContext);
+                }
+
+                foreach (var relationshipField in part.PartDefinition.Fields)
+                {
+                    itemContext.SetContentPartFieldDefinition(relationshipField);
+                    itemContext.SetContentField((JObject)contentItem.Content[relationshipField.PartDefinition.Name][relationshipField.Name]);
+
+                    var fieldSyncer = _contentFieldsGraphSyncers.FirstOrDefault(x => x.FieldTypeName == relationshipField.FieldDefinition.Name);
```

Comment:
Field handling should be delegated to the part syncer (and the field helper), e.g. see `EponymousPartGraphSyncer`/`FlowPartGraphSyncer` (and `ContentFieldsGraphSyncer`).